### PR TITLE
Fixed a bug in #4199 that resulted in wrong Out of Sync.

### PR DESCRIPTION
**Pull request by funera1**
**What this PR does / why we need it**:
Fixed a bug in #4199 that resulted in wrong Out of Sync.
Also, during this modification, the diff option WithIgnoreAddingMapKeys was made commutative. That is, diff a b = diff b a.
I would like to discuss the validity of this modification.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by funera1** [comment]
@knanao 
Please review this PR, when you have time:bow:

**Review by knanao — APPROVED** [review]

File: pkg/app/piped/platformprovider/kubernetes/diff.go [hunk 1]
```diff
@@ -53,19 +53,19 @@ type DiffListChange struct {
 func Diff(old, new Manifest, logger *zap.Logger, opts ...diff.Option) (*diff.Result, error) {
 	if old.Key.IsSecret() && new.Key.IsSecret() {
 		var err error
-		new.u, err = normalizeNewSecret(old.u, new.u)
+		old.u, err = normalizeNewSecret(old.u, new.u)
```
Comment (khanhtc1202) [inline pkg/app/piped/platformprovider/kubernetes/diff.go:56]: sorry but why does `normalizeNewSecret` return `old` stub 🤔 This does not make sense to me 👀

Comment (funera1) [reply pkg/app/piped/platformprovider/kubernetes/diff.go:56]: In a nutshell, it is because we swapped the order of headManifest and liveManifest in Diff.
ignoreAddingMapKeys in diff/diff.go is to ignore LiveManifestw and normalizeNewSecret depends on to ignore LiveManifest with ignoreAddingMapKeys!v

Comment (khanhtc1202) [reply pkg/app/piped/platformprovider/kubernetes/diff.go:56]: Thanks for clarifying, then in that case, why not just mention directly here that it's Live/Head manifest or, even better Base/Compare instead of Old/New one here? 🤔 I'm not talking about the logic but the readability of the code, return value from `normalizeNewSecret` is set to `old` one doesn't make sense to me, that's my point 👀

Comment (funera1) [reply pkg/app/piped/platformprovider/kubernetes/diff.go:56]: I see what you mean. I too have a problem with the fact that old/new is not explicitly tied to live/head.
However, since the relationship between old/new and live/head is not only in this section, but is tied across the board, I also think that a separate Issue or PR should be created to rename it.

Or is your point that normalizeNewSecret should be normalizeOldSecret? If so, I will create an improved PR immediately.

Comment (khanhtc1202) [reply pkg/app/piped/platformprovider/kubernetes/diff.go:56]: >  I also think that a separate Issue or PR should be created to rename it.

Yes, It's what I mean too 👍 The logic around this part is tricky, and if the flow & naming are hard to follow, it's hard for us to debug next time problems occur around here.

Comment (funera1) [reply pkg/app/piped/platformprovider/kubernetes/diff.go:56]: Ok, I will create this issue. Thank you for nice comment:smile:
